src/core/include/openvino/core/model.hpp

```diff
@@ -42,7 +42,7 @@ class ModelAccessor;
  * @brief A user-defined model
  * @ingroup ov_model_cpp_api
  */
-class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
+class OPENVINO_API Model : public std::enable_shared_from_this<Model>, public IRtti {
```

Comment:
why do we need this inheritance ? I suppose we can hide all the details in OPENVINO_RTTI_BASE

Reply:
That was my initial approach in first commit https://github.com/openvinotoolkit/openvino/pull/28911/commits/f6096002894f74643752a773178af5c1a3599c40, but it failed to suppress `suggest-override` warning on clang. So I introduced `IRtti` to solve override conflicts.

Reply:
> suggest-override

we can suppress directly in C++ code using `OPENVINO_DO_PRAGMA`

Reply:
Done.